test/src/test_derive.rs

```diff
@@ -160,3 +167,181 @@ fn test_derive_owned_to_variant() -> bool {
 
     ok
 }
+
+#[derive(gdnative::NativeClass)]
+#[inherit(Node)]
+struct FakeNativeClassWithPropertyHooks {
+    #[property(before_set = "Self::before_get_fake_callback")]
+    before_get_bool: bool,
+    #[property(after_set = "Self::after_get_fake_callback")]
+    after_get_bool: bool,
+    #[property(before_set = "Self::before_set_fake_callback")]
+    before_set_bool: bool,
+    #[property(after_set = "Self::after_set_fake_callback")]
+    after_set_bool: bool,
+
+    pub before_get_callback_calls: u32,
+    pub after_get_callback_calls: u32,
+    pub before_set_callback_calls: u32,
+    pub after_set_callback_calls: u32,
+}
+
+#[gdnative_derive::methods]
+impl FakeNativeClassWithPropertyHooks {
+    fn new(_owner: &Node) -> Self {
+        Self {
+            before_get_bool: false,
+            after_get_bool: false,
+            before_set_bool: false,
+            after_set_bool: false,
+            before_get_callback_calls: 0,
+            after_get_callback_calls: 0,
+            before_set_callback_calls: 0,
+            after_set_callback_calls: 0,
+        }
+    }
+
+    fn before_get_fake_callback(_script: &mut Self, _owner: TRef<Node, Shared>) {
+        _script.before_get_callback_calls += 1;
+    }
+
+    fn after_get_fake_callback(_script: &mut Self, _owner: TRef<Node, Shared>) {
+        _script.after_get_callback_calls += 1;
+    }
+
+    fn before_set_fake_callback(_script: &mut Self, _owner: TRef<Node, Shared>) {
+        _script.before_set_callback_calls += 1;
+    }
+
+    fn after_set_fake_callback(_script: &mut Self, _owner: TRef<Node, Shared>) {
+        _script.after_set_callback_calls += 1;
+    }
+}
+
+fn test_derive_nativeclass_with_property() -> bool {
+    println!(" -- test_derive_nativeclass_with_property");
+
+    #[derive(gdnative::NativeClass)]
+    #[inherit(Node)]
+    struct FakeNativeClass {
+        #[property]
+        pub simple_bool: bool,
+    }
+    impl FakeNativeClass {
+        fn new(_owner: &Node) -> Self {
+            Self { simple_bool: false }
+        }
+    }
+
+    let ok = std::panic::catch_unwind(|| {
+        let owner = Node::new();
+        let class = FakeNativeClass::new(&owner);
+        assert_eq!(class.simple_bool, false);
+    })
+    .is_ok();
+
+    if !ok {
+        gdnative::godot_error!("   !! Test test_derive_nativeclass_with_property failed");
+    }
+
+    ok
+}
```

Comment:
This test is useless because it's simply accessing the Rust field, not through GDNative. `#[property]` without any hooks is a pretty trivial case and I don't think it really needs to be tested, so you may also remove it outright.

Reply:
My theory here was testing this built a foundation that allowed me to trust the rest of the testing. What's the harm in keeping it? In fact, during my dev, a similar manually executed test caught errors I had a made a number of times. I would prefer to keep it for the coverage. Having said that, if you are strongly opposed, I can remove it.

Reply:
Did you read the comment? I said that it doesn't test what you think it's testing. What is tested here by this code is just rustc's ability to calculate field offsets, and not the implementation of `#[property]` **at all**. If you want to keep it, then at least use `Instance` and `Object::set`.

Reply:
I did read your comment, and I did understand it. What I was testing was that the property didn't disappear because of the #[property]. I agree that using the something like a set and a get would be better, so I will add that.

FWIW, your response reads like you're annoyed. I am sorry if I annoyed you. I just want a good quality change. And I appreciate your help.